PlayerPanel.js

```diff
@@ -34,87 +34,136 @@ function update_pclist() {
-
-		row_inside.append(conditions);
 
+		let playerData;
 		if (pc.sheet in window.PLAYER_STATS) {
-			data = window.PLAYER_STATS[pc.sheet];
-
-			stats.find(".hp").text(data.hp + "/" + data.max_hp);
-			stats.find(".ac").text(data.ac);
-			stats.find(".pp").text(data.pp);
-
-			for (var i = 0; i < data.conditions.length; i++)
-				conditions.find(".conditions").append($("<div/>").text(data.conditions[i]));
+			playerData = window.PLAYER_STATS[pc.sheet];
 		}
 
+		const newPlayerTemplate = `
+			<div class="player-card">
+				<div class="player-card-header">
+					<div class="player-name">${pc.name}</div>
+					<div class="player-actions">
+						<button 
+							data-set-token-id='${pc.sheet}' data-img='${pc.image}' data-name="${pc.name}"
+							data-hp="${playerData ? playerData.hp : ''}"
+							data-ac="${playerData ? playerData.ac : ''}"
+							data-maxhp="${playerData ? playerData.max_hp : ''}"
+							data-color="${color}" class="add-token-btn">
+							TOKEN
+						</button>
+						<button data-target='${pc.sheet}' class="open-sheet-btn">SHEET</button>
+					</div>
+				</div>
+				<div class="player-card-content">
+					<div class="player-token" style="box-shadow: ${
+						playerData ? `${token_health_aura(
+							Math.round((playerData.hp / playerData.max_hp) * 100)
+						)} 0px 0px 11px 3px` : 'none'
+					};">
+						<img width="70" height="70" src="${pc.image}" style="border: 2px solid ${color}" />
+						${
+							playerData ? `
+								<div class="player-token-hp">${playerData.hp} / ${playerData.max_hp}</div>
+								<div class="player-token-ac">${playerData.ac}</div>
+							` : ''
+						}
+					</div>
+					${
+						playerData ? `
+							<div class="player-info">
+								<div class="player-attributes">
+									<div class="player-attribute">
+										<img src="${window.EXTENSION_PATH + "assets/eye.png"}" title="Passive Perception" />
+										<span>${playerData.pp}</span>
+									</div>
+									<div class="player-attribute">
+									<img src="${window.EXTENSION_PATH + "assets/walking.png"}" title="Walking Speed" />
+									<span>${playerData.walking}</span>
+									</div>
+									<div class="player-attribute">
+										<img src="${window.EXTENSION_PATH}assets/inspiration.svg" title="Inspiration" />
+										<span>${playerData.inspiration ? 'Yes' : 'No'}</span>
+									</div>
+								</div>
+								<div class="player-conditions">
+									<div class="player-card-title"><b>Conditions:</b></div>
+									<div>
+										${
+											playerData.conditions.map(c => `<span title="${
+												CONDITIONS[c] ? [c, ...CONDITIONS[c]].join(`\n`) : [c, ...CONDITIONS.Exhaustion].join(`\n`)
+											}">${c}</span>`).join(', ')
+										}
+									</div>
+								</div>
+							</div>
+						` : `
+							<div class="player-no-attributes">
+								Please load the character sheet.
+							</div>
+						`
+					}
+				</div>
+				${
+					playerData ? `

```

Comment:
In 0.0.41 I fixed this by changing to "playerData && playerData.abilities".
I backported that to main. Base your branch on main and I'll pull.

Reply:
done!